lib/presentation/widget/shared_space_details/shared_space_details_widget.dart

```diff
@@ -322,4 +325,16 @@ class _SharedSpaceDetailsWidgetState extends State<SharedSpaceDetailsWidget> {
           ],
         )));
   }
+
+  void selectRoleBottomSheet(
+      BuildContext context,
+      SharedSpaceRoleName selectedRole,
+      Function(SharedSpaceRoleName) onSelectedRoleCallback) {
```

Comment:
please strong typing this with `typeDef`

Reply:
It's already defined strong type: 
```
typedef OnSelectActionClick = void Function(SharedSpaceRoleName role);
```

Reply:
why not place it here?

Reply:
IMO, the typedef should define in the isolated module. In there, it's `SelectRoleModalSheetBuilder`. What happens if there are many many of typedef in the widget? The widget should be the usage of it only, should not define other things in the widget.